service/src/error.rs

```diff
@@ -6,6 +6,8 @@ use thiserror::Error;
 pub enum Error {
     #[error("IO error: {0}")]
     Io(#[from] std::io::Error),
+    #[error("Bincode error: {0}")]
+    Bincode(#[from] bincode::Error),
```

Comment:
This tends to give FFI crashes when `bincode` will be behind a features flag. It will be better to create a new Error type in `persist.rs`